diesel/src/macros/mod.rs

```diff
@@ -228,108 +252,248 @@ macro_rules! __diesel_column {
 /// ```
 #[macro_export]
 macro_rules! table {
-    // Put `use` statements at the end because macro_rules! cannot figure out
-    // if `use` is an ident or not (hint: It's not)
+    // Put imports into the import field
     (
-        use $($import:tt)::+; $($rest:tt)+
+        @parse
+        import = [$(use $($import:tt)::+;)*];
+        doc = [$($doc:expr,)*];
```

Comment:
Do we need to match on doc this early? If nothing else it seems like this could just be `doc = $doc:tt` -- but it seems like we could just not stick the `doc = []` until we get to that point. Also not sure why we're collecting `doc` into a repeater, but I guess I'll find out.